GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -298,12 +304,124 @@ public ArgumentString GetRevisionFilter()
 
             ArgumentBuilder filter = new();
 
+            // Separate the filters in groups
+            GetCommitRevisionFilter(filter);
+            GetLimitingRevisionFilter(filter);
+            GetBranchRevisionFilter(filter);
+
+            return filter;
+        }
+
+        public string GetSummary()
+        {
+            StringBuilder filter = new();
+
+            GetCommitFilterSummary(filter);
+            GetLimitingFilterSummary(filter);
+            GetBranchFilterSummary(filter);
+
+            return filter.ToString();
+        }
+
+        /// <summary>
+        /// Internal grouping of filter arguments.
+        /// Commit limiting filters, normally not affecting parents.
+        /// </summary>
+        /// <param name="filter">ArgumentBuilder arg</param>
+        private void GetCommitRevisionFilter(ArgumentBuilder filter)
+        {
             if (CommitsLimit > 0)
             {
                 filter.Add($"--max-count={CommitsLimit}");
             }
 
-            // Branch filters
+            if (ByDateFrom)
+            {
+                filter.Add($"--since=\"{DateFrom:yyyy-MM-dd hh:mm:ss}\"");
+            }
+
+            if (ByDateTo)
+            {
+                filter.Add($"--until=\"{DateTo:yyyy-MM-dd hh:mm:ss}\"");
+            }
+        }
+
+        /// <summary>
+        /// Internal grouping of filter arguments.
+        /// Commit limiting filters (affecting parents).
+        /// </summary>
+        /// <param name="filter">ArgumentBuilder arg</param>
+        private void GetLimitingRevisionFilter(ArgumentBuilder filter)
+        {
+            if (!ShowMergeCommits)
+            {
+                filter.Add("--no-merges");
+            }
+
+            // Listed in Git help as history simplification, but is a revision filter
+            if (ShowSimplifyByDecoration)
+            {
+                filter.Add("--simplify-by-decoration");
+            }
+
+            if (ByAuthor && !string.IsNullOrWhiteSpace(Author))
+            {
+                filter.Add($"--author=\"{Author}\"");
+            }
+
+            if (ByCommitter && !string.IsNullOrWhiteSpace(Committer))
+            {
+                filter.Add($"--committer=\"{Committer}\"");
+            }
+
+            if (IgnoreCase && (ByAuthor || ByCommitter || ByMessage || ByDiffContent))
+            {
+                filter.Add("--regexp-ignore-case");
+            }
+
+            if (ByDiffContent && !string.IsNullOrWhiteSpace(DiffContent))
+            {
+                filter.Add($"-G\"{DiffContent}\"");
+            }
+
+            if (ByMessage && !string.IsNullOrWhiteSpace(Message))
+            {
+                if (Message.Trim().StartsWithAny(_messageAsGitOptions))
+                {
+                    filter.Add(Message);
+                }
+                else
+                {
+                    filter.Add($"--grep=\"{Message}\"");
+                }
```

Comment:
```suggestion
                string message = Message.TrimStart();
                if (message.StartsWithAny(_messageAsGitOptions))
                {
                    filter.Add(message);
                }
                else
                {
                    // The user may want to grep with leading or trailing whitespaces, retain the original message
                    filter.Add($"--grep=\"{Message}\"");
                }
```

Reply:
Should probably apply the same to the similar code below.

Reply:
Trim() applied where info is read from the filter boxes.